### ISSUE TITLE
Convert 'px' values to 'rem' to ensure the components appear correctly when using different browser font sizes

### ISSUE THREAD

**Issue by GCHQ-Developer-847** [mi6/ic-ui-kit#290]
## Summary
A lot of spacing and sizing values in the CSS for the components use the `px` unit. This means a lot of the components render incorrectly when the font size in the user's browser is changed to something other than the default. Using `rem` instead would make the components resize and render correctly at different font sizes.

## 💬 Description
Users can change the font size in their browser e.g. Settings > Appearance > Font size in Chrome, or by pressing Option -Command - Plus (+) in Safari.

Doing so causes a lot of different parts of the components to display incorrectly, such as icons appearing too large or small, or elements not being aligned with each other. 

For example, at the 'Very large' font size on Chrome:
![image](https://user-images.githubusercontent.com/111882109/212361809-70917dd2-94d3-49f2-a7e1-58fd72167eb6.png)
![image](https://user-images.githubusercontent.com/111882109/212364255-9a8fb5d0-8226-4f23-92e8-8108b99229df.png)
![image](https://user-images.githubusercontent.com/111882109/212362050-19885b6d-d979-4638-835e-05890c846c5b.png)

After some exploration, it appears that the cause is using the absolute `px` unit for a lot of the spacing and sizing on components - `rem` is a relative unit so using this for elements will make them scale to match the chosen font size.

Some of the spacing and a lot of the text used in components seem to be ok as they use the CSS spacing and typography variables which already have `rem` values, but there is still a lot of other styling which uses `px` values.

Most of the components are currently affected by this issue. 

**Note 1:** I believe it should be almost all `px` values that need changing across all components. Anything to do with breakpoints (e.g. the --ic-breakpoint-xs etc. variables, media queries) would need to remain, but I would expect that every other `px` value would need converting.

**Note 2:** Converting `px` to `rem` _should_ fix all issues that occur when using a different font size, but it will be important to double-check all of the components after doing this to make sure there aren't any other changes that need to be made to make them appear correctly at different font sizes.

This is a useful website for understanding more about `rem`: [What Are Rem Units? (& How to Use Them in CSS)](https://blog.hubspot.com/website/css-rem).

## 💰 Use value
This is important from an accessibility perspective as many users may want to set their browser font size to one that is different from the default to make information easier for them to read. The usability of a lot the components will be affected if they don't accommodate a different font size by resizing accordingly.
